modules/network/src/main/java/org/apache/ignite/network/scalecube/ScaleCubeClusterServiceFactory.java

```diff
@@ -159,7 +168,12 @@ public void stop() {
                 }
 
                 cluster.shutdown();
-                cluster.onShutdown().block();
+
+                try {
+                    shutdownFuture.get(10, TimeUnit.SECONDS);
```

Comment:
Why is it 10 seconds? Is it some estimate, or just a value that seems sufficient?

Reply:
Just a value that seems sufficient. Do you have any other ideas?

Reply:
The shutdown procedure leaves the cluster, shuts down membership, gossip, metadata store and does not seem to wait for any long-running tasks (or executors, if I didn't miss anything). So 10 seconds seems to be a good start.